specifyweb/frontend/js_src/lib/components/DataModel/businessRuleDefs.ts

```diff
@@ -165,6 +169,37 @@ export const businessRuleDefs: MappedBusinessRuleDefs = {
                 .then(async (accepted) =>
                   accepted === null ? taxon : getLastAccepted(accepted)
                 );
+
+            const collectionObject = determination.collection
+              ?.related as SpecifyResource<CollectionObject>;
+            collectionObject
+              .rgetPromise('collectionObjectType', true)
+              .then((coType: SpecifyResource<CollectionObjectType>) => {
```

Comment:
The `coType` can technically be `null` here, such as when the `collectionObjectType` is not yet set on the CollectionObject. 
(Not sure how we want to handle that case, but worthy a mention). 

Also, it is still possible to have an invalid `Determination -> taxon` here.
This Business Rule is only ran when the Determination changes. The user can change insert the Determination first, and _then_ change the `CollectionObejct -> coType` which uses a different TaxonTreeDef compared to the `Determination -> Taxon`.

Reply:
`collectionObjectType` is set as a required field in the schema due to which it becomes a required field in the form as well, even if you set `isrequired=false` in the form definition. So ideally there would always be a `collectionObjectType` - especially if the plan is to set a default based on discipline?

#5100 should handle the case when `collectionObjectType` is changed after selecting a `Determination -> taxon`.

Reply:
> So ideally there would always be a collectionObjectType

I agree that for existing CollectionObjects, collectionObjectType will always have a value. 
I specifically meant for new CollectionObjects which have not yet selected a `collectionObjectType` yet, the value will always be null/undefined. 

In this case, they can set a `Determination -> taxon` on the CollectionObject before they select a CollectionObjectType. 
We can let #5100 handle that case if we want. 

I think the intention was to have the CollectionObjectType on the CollectionObject default to whatever the Collection -> CollectionObjectType is. 

In that case, yes it would be another business rule (😵) to set that default, if we want to autofill the type on the CollectionObject.
(Either let, or always have the user explicitly select which CollectionObjectType to use. Whichever approach fits more with our requirements).

Reply:
Setting default cotype with business rule should be super straightforward. When creating a new CO:
- Default COType will be set already (maybe a different PR for that in case we go with scoping QCBX instead of this PR?)
- Determination -> taxon will be validated to make sure taxon treeDef is same as coType treeDef
- Changing COType after that will clear all fields (won't clear coType) due to #5100 

Sounds good to me.